nomos-core/chain-defs/src/utils/merkle.rs

```diff
@@ -13,3 +18,112 @@ impl<T> MerkleNode<T> {
 }
 
 pub type MerklePath<T> = Vec<MerkleNode<T>>;
+
+pub fn calculate_merkle_root<T>(leaves: &[MerkleNode<T>]) -> Option<Fr>
```

Comment:
a `MerkleNode` is meant to be for internal nodes, not leaves, we can just ask for `T`s here